Generator emits optional parameters before required ones (CS1737)

## Problem

The C# code generator emits query parameters in the order they appear in the OpenAPI spec, without sorting by requiredness. When an optional parameter appears before a required one in the spec, the generated method signature violates C#'s rule that required parameters must precede optional ones.

**Build error:**
```
error CS1737: Optional parameters must appear after all required parameters
```

## Reproduction

```bash
SPEC_REF=stable/8.8 bash scripts/build.sh
```

Two methods fail to compile:
- `CreateDocumentLinkAsync(..., string? storeId = null, string contentHash, ...)`
- `GetDocumentAsync(..., string? storeId = null, string contentHash, ...)`

Both have `storeId` (optional, `required: false`) listed before `contentHash` (required, `required: true`) in the OpenAPI spec.

## Why it's latent on main

On `stable/8.9`, `contentHash` was relaxed to `required: false`, so both parameters are optional and ordering doesn't matter. The bug only surfaces when a spec has mixed required/optional query parameters in an order that puts optional before required.

## Expected behavior

The generator should sort method parameters so that all required parameters come before optional ones, regardless of their order in the OpenAPI spec.

## Affected component

`src/Camunda.Orchestration.Sdk.Generator/CSharpClientGenerator.cs` — the method signature emission logic for query parameters.

--- a/external-spec/bundled/rest-api.bundle.json
+++ b/external-spec/bundled/rest-api.bundle.json
@@ -7185,7 +7185,7 @@
         ],
         "operationId": "getGlobalJobStatistics",
         "summary": "Global job statistics",
-        "description": "Returns global aggregated counts for jobs. Optionally filter by the creation time window and/or jobType.\n",
+        "description": "Returns global aggregated counts for jobs. Filter by the creation time window (required) and optionally by jobType.\n",
         "parameters": [
           {
             "name": "from",

--- a/src/Camunda.Orchestration.Sdk.Generator/CSharpClientGenerator.cs
+++ b/src/Camunda.Orchestration.Sdk.Generator/CSharpClientGenerator.cs
@@ -90,6 +90,7 @@ internal static class CSharpClientGenerator
                     .ToList();
                 var queryParams = op.Parameters
                     .Where(p => p.In == ParameterLocation.Query)
+                    .OrderByDescending(p => p.Required)
                     .Select(p => new ParamMeta(p.Name, MapType(p.Schema), p.Required))
                     .ToList();
 

--- a/src/Camunda.Orchestration.Sdk/Generated/CamundaClient.Generated.cs
+++ b/src/Camunda.Orchestration.Sdk/Generated/CamundaClient.Generated.cs
@@ -1894,7 +1894,7 @@ public partial class CamundaClient
 
     /// <summary>
     /// Global job statistics
-    /// Returns global aggregated counts for jobs. Optionally filter by the creation time window and/or jobType.
+    /// Returns global aggregated counts for jobs. Filter by the creation time window (required) and optionally by jobType.
     /// 
     /// </summary>
     /// <remarks>Operation: getGlobalJobStatistics</remarks>

--- a/src/Camunda.Orchestration.Sdk/Generated/SpecHash.Generated.cs
+++ b/src/Camunda.Orchestration.Sdk/Generated/SpecHash.Generated.cs
@@ -10,5 +10,5 @@ public partial class CamundaClient
     /// <summary>
     /// SHA-256 digest of the OpenAPI spec this SDK was generated from.
     /// </summary>
-    public const string SpecHash = "sha256:ce28935d7c5d940ac3249eca6c65a20aaa50286c20c365ad00deb315534d2a16";
+    public const string SpecHash = "sha256:5de81e6b2d3f152265e6f6f042a776f0bc2ca73572e9f5a8c54eee460f5c6161";
 }

--- a/test/Camunda.Orchestration.Sdk.Tests/ParameterOrderingTests.cs
+++ b/test/Camunda.Orchestration.Sdk.Tests/ParameterOrderingTests.cs
@@ -1,0 +1,43 @@
+using System.Reflection;
+
+namespace Camunda.Orchestration.Sdk.Tests;
+
+/// <summary>
+/// Regression test: generated methods must not have optional parameters
+/// before required ones (CS1737). This validates the entire CamundaClient
+/// surface, not a single method — catching any future spec ordering issues.
+/// </summary>
+public class ParameterOrderingTests
+{
+    [Fact]
+    public void AllPublicMethods_HaveRequiredParamsBeforeOptional()
+    {
+        var methods = typeof(CamundaClient)
+            .GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly)
+            .Where(m => !m.IsSpecialName); // exclude property getters/setters
+
+        var violations = new List<string>();
+
+        foreach (var method in methods)
+        {
+            var parameters = method.GetParameters();
+            var seenOptional = false;
+
+            foreach (var param in parameters)
+            {
+                if (param.HasDefaultValue || param.IsOptional)
+                {
+                    seenOptional = true;
+                }
+                else if (seenOptional)
+                {
+                    violations.Add(
+                        $"{method.Name}: required parameter '{param.Name}' appears after optional parameter");
+                    break;
+                }
+            }
+        }
+
+        Assert.Empty(violations);
+    }
+}

